core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorTableHandle.java

```diff
@@ -13,10 +13,29 @@
  */
 package io.trino.spi.connector;
 
+import java.util.Random;
+
 /**
  * Represents a handle to a relation returned from the connector to the engine.
  * It will be used by the engine whenever given relation will be accessed.
  */
 public interface ConnectorTableHandle
 {
+    /**
+     * Returns true if a ConnectorTableHandle is "fusible" with another ConnectorTableHandler.
+     * Fusible here means the two tables can possibly be optimized by SharedScanOptimizer.
+     */
+    default boolean equalsForFusion(Object o)
+    {
+        return false;
+    }
+
+    /**
+     * Returns the hashcode that is & should only be used by Query Fusion to compare the table handles and group them into HashMaps.
+     */
+    default int hashCodeForFusion()
+    {
+        Random rand = new Random();
+        return rand.nextInt();
```

Comment:
This seems wrong as a default implementation. The hash code for an object should never change. This would be a better default:
```java
return 0;

Reply:
As you mentioned, this is not a standard equals method. The intention is to **not** have the same hash code so that they are not fusible from the perspective of this specific rule. It only becomes useful when a specific ConnectorTableHandle creates its own implementation.